Signal/src/ViewControllers/FingerprintViewController.m

```diff
@@ -19,26 +21,41 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-@interface FingerprintViewController () <OWSHighlightableLabelDelegate, OWSCompareSafetyNumbersActivityDelegate>
+typedef void (^CustomLayoutBlock)();
+
+@interface CustomLayoutView : UIView
```

Comment:
There are many situations where iOS auto layout becomes awkward.  Some of them can get solved using constraint priority, compression resistance and content hugging but in practice I find this makes for unmaintainable, bug-prone code.  One simple solution for some of these cases is to use manual, block-based layout.  I'll probably eventually pull this out into its own source file.